apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/CreateFirebaseAuthDialog.tsx

```diff
@@ -0,0 +1,170 @@
+    defaultValues: {
+      enabled: true,
+      firebaseProjectId: '',
+    },
+  })
+
+  useEffect(() => {
+    if (visible) {
+      form.reset({
+        enabled: true,
+        firebaseProjectId: '',
+      })
+      // the form input doesn't exist when the form is reset
+      setTimeout(() => {
+        form.setFocus('firebaseProjectId')
+      }, 25)
+    }
+  }, [visible])
+
+  const onSubmit: SubmitHandler<z.infer<typeof FormSchema>> = async (values) => {
+    createAuthIntegration({
+      projectRef: projectRef!,
+      oidcIssuerUrl: `https://securetoken.google.com/${values.firebaseProjectId}`,
+    })
+  }
+
+  return (
+    <Dialog open={visible} onOpenChange={() => onClose()}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle className="truncate">
+            {isCreating
+              ? `Add new Firebase Auth connection`
+              : `Update existing Firebase Auth connection`}
+          </DialogTitle>
+        </DialogHeader>
+
+        <Separator />
+        <DialogSection>
+          <Form_Shadcn_ {...form}>
+            <form id={FORM_ID} onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
+              {/* Enabled flag can't be changed for now because there's no update API call for integrations */}
+              {/* <FormField_Shadcn_
+              key="enabled"
+              control={form.control}
+              name="enabled"
+              render={({ field }) => (
+                <FormItemLayout
+                  className="px-8"
+                  label={`Enable Firebase Auth Connection`}
+                  layout="flex"
+                >
+                  <FormControl_Shadcn_>
+                    <Switch
+                      checked={field.value}
+                      onCheckedChange={field.onChange}
+                      disabled={field.disabled}
+                    />
+                  </FormControl_Shadcn_>
+                </FormItemLayout>
+              )}
+            />
+            <Separator /> */}
+
+              <p className="text-sm text-foreground-light">
+                This will enable a JWT token from a specific Firebase project to access data from
+                this Supabase project.
+              </p>
+              <FormField_Shadcn_
+                key="firebaseProjectId"
+                control={form.control}
+                name="firebaseProjectId"
+                render={({ field }) => (
+                  <FormItemLayout label="Firebase Auth Project ID">
+                    <FormControl_Shadcn_>
+                      <Input_Shadcn_ {...field} />
+                    </FormControl_Shadcn_>
+                  </FormItemLayout>
+                )}
+              />
+
+              <InformationBox
+                title="Note: This connection requires a Row Level Security (RLS) policy"
+                description="You will need to manually insert the policy after creating this 3rd party Auth connection."
+              />
```

Comment:
should we just use `<Alert_Shadcn />` here? i feel like this info is needed and shouldn't be hidden

Reply:
also can probably remove the "Note:"
```suggestion
              <InformationBox
                title="This connection requires a Row Level Security (RLS) policy"
                description="You will need to manually insert the policy after creating this 3rd party Auth connection."
              />
```

Reply:
also - might be helpful to explain why an RLS policy is needed? from the pov of someone looking at this with a fresh pair of eyes, im wondering why a policy is required here

Reply:
Fixed.